pkg/cli/alpha/config-gen/cmd.go

```diff
@@ -0,0 +1,278 @@
+		},
+	}.GetCommand()
+
+	if os.Getenv("KUSTOMIZE_FUNCTION") == "true" {
+		// run as part of kustomize -- read from stdin
+		c.Args = cobra.MinimumNArgs(0)
+	} else {
+		c.Args = cobra.MinimumNArgs(1)
+	}
+	c.RemoveCommand(c.Commands()...)
+	c.Use = "config-gen PROJECT_FILE [RESOURCE_PATCHES...]"
+	c.Version = `v1alpha1`
+	c.Short = `Generate configuration for a controller-runtime based projects`
+	c.Long = strings.TrimSpace(`
+config-gen programatically generates configuration for a controller-runtime based
+project using the project source code (golang) and a KubebuilderConfigGen resource file.
+
+This is an alternative to expressing configuration as a static set of kustomize patches
+in the "config" directory. 
+
+config-gen may be used as a standalone command run against a file, as a kustomize
+transformer plugin, or as a configuration function (e.g. kpt).
+
+config-gen uses the controller-tools generators to generate CRDs from the go source
+and then generates additional resources such as the namespace, controller-manager,
+webhooks, etc.
+
+Following is an example KubebuilderConfigGen resource used by config-gen:
+
+  # kubebuilderconfiggen.yaml
+  # this resource describes how to generate configuration for a controller-runtime
+  # based project
+  apiVersion: kubebuilder.sigs.k8s.io
+  kind: KubebuilderConfigGen
+  metadata:
+    name: project-name
+  spec:
+    image: org-name/project-name:v0.1.0
+
+If this file was at the project root, config-gen could be used as:
+
+  kubebuilder alpha config-gen ./kubebuilderconfiggen.yaml
+
+The KubebuilderConfigGen resource has the following fields:
+
+  apiVersion: kubebuilder.sigs.k8s.io
+  kind: KubebuilderConfigGen
+  metadata:
+    # name of the project
+	# required
+    name: project-name
+
+	# namespace for the project
+	# optional -- defaults to NAME-system
+	namespace: project-namespace
+
+  spec:
+    # image used to run the controller-manager in the Deployment
+	# required
+	image: org/project:v0.1.0
+
+    # path to project root directory to generate the CRDs
+	# optional -- defaults to '.'
+    directory: ./relative/path
+
+	# enable / disable generation of specific components
+	# optional -- defaults to the values shown
+	components:
+	  controller-manager: true
+	  crds: true
+	  namespace: true
+	  rbac: true
+	  cert-manager: false
+	  prometheus: false
+	  webhooks: false
+
+	# disable using the auth proxy
+	# optional -- defaults to false
+	disableAuthProxy: false
+    
+	# enable conversion webhooks for a particular API
+	# requires setting the webhook component to true
+	# optional -- defaults to empty
+	conversionWebhooks:
+	  "bars.example.my.domain": true
+
+	# if set, use component config for the project
+	# optional -- defaults to unset
+	componentConfigFilepath: ./path/to/componentconfig.yaml
+
+	# options for project development not intended for production
+	developmentOptions:
+	  # generate the webhook certificate and wire it as needed
+	  generateCert: true
```

Comment:
You've got mixed spaces and tabs in this example what makes it hard to read.

Reply:
Done.  Good catch